core/templates/domain/objects/NumberWithUnitsObjectFactorySpec.ts

```diff
@@ -176,5 +191,54 @@ describe('NumberWithUnitsObjectFactory', () => {
         'SyntaxError: In "kg / m^(2)", "^" must be ' +
         'followed by a floating-point number');
     });
+
+    it('should create currency units', () => {
+      const createCurrencyUnitsSpy = spyOn(
+        nwuof.unitsFactory, 'createCurrencyUnits');
+      nwuof.createCurrencyUnits();
+      expect(createCurrencyUnitsSpy).toHaveBeenCalled();
+    });
+
+    it('should create NumberWithUnits object from dict', () => {
+      let numberWithUnitsObject = {
+        type: 'dummy-type',
+        real: 1,
+        fraction: {
+          isNegative: false,
+          wholeNumber: 2,
+          numerator: 1,
+          denominator: 3
+        },
+        units: [
+          {
+            unit: 'Kg',
```

Comment:
Isn't kg normally spelled with a small letter k? If so, then let's change this and the following accordingly.

Reply:
Done